packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -249,50 +249,58 @@ abstract class ProcessUtils {
     required String line,
     required void Function(Object, StackTrace) onError,
   }) async {
-    final Completer<void> completer = Completer<void>();
-
-    void writeFlushAndComplete() {
-      stdin.writeln(line);
-      stdin.flush().whenComplete(() {
-        if (!completer.isCompleted) {
-          completer.complete();
-        }
-      });
-    }
-
-    runZonedGuarded(
-      writeFlushAndComplete,
-      (Object error, StackTrace stackTrace) {
-        onError(error, stackTrace);
-        if (!completer.isCompleted) {
-          completer.complete();
-        }
-      },
+    await _writeToStdinGuarded(
+      stdin: stdin,
+      content: line,
+      onError: onError,
+      isLine: true,
     );
-    return completer.future;
   }
 
   /// See [writelnToStdinGuarded]. This calls `stdin.write` instead of `stdin.writeln`.
   static Future<void> writeToStdinGuarded({
     required IOSink stdin,
-    required String line,
+    required String content,
+    required void Function(Object, StackTrace) onError,
+  }) async {
+    await _writeToStdinGuarded(
+      stdin: stdin,
+      content: content,
+      onError: onError,
+      isLine: false,
+    );
+  }
+
+  static Future<void> _writeToStdinGuarded({
+    required IOSink stdin,
+    required String content,
     required void Function(Object, StackTrace) onError,
+    required bool isLine
   }) async {
     final Completer<void> completer = Completer<void>();
 
     void writeFlushAndComplete() {
-      stdin.write(line);
-      stdin.flush().whenComplete(() {
-        if (!completer.isCompleted) {
-          completer.complete();
-        }
+      if (isLine) {
+        stdin.writeln(content);
+      } else {
+        stdin.write(content);
+      }
+      stdin.flush().then((_) {
+        completer.complete();
       });
     }
 
     runZonedGuarded(
       writeFlushAndComplete,
       (Object error, StackTrace stackTrace) {
-        onError(error, stackTrace);
+        try {
+          onError(error, stackTrace);
+        }
+        on Exception catch (e) {
```

Comment:
nit. put this on previous line after close curly bracket

Reply:
Thanks! Addressed in [nits](https://github.com/flutter/flutter/pull/150120/commits/f728c1562a00285b05432a61fd8a4d7eb1bfa329)